src/test/java/plugins/GitPluginTest.java

```diff
@@ -371,6 +372,299 @@ public void inverse_strategy_to_choose_build() {
+        );
+    }
+
+    @Test
+    public void commit_author_in_changelog_test() {
+        URL changesUrl;
+        final String TEST_COMMIT_MESSAGE = "Second commit";
+        final String BRANCH_NAME = "testBranch";
+        final String TEST_FILE = "foo_test_branch";
+
+        Build b;
+        GitRepo repo = buildGitRepo();
+
+        repo.createBranch(BRANCH_NAME);
+        repo.touch(TEST_FILE);
+        repo.git("add", TEST_FILE);
+        repo.git("commit", "-m", TEST_COMMIT_MESSAGE, "--author", "New-Author <other-email@example.org>");
+        repo.transferToDockerContainer(host, port);
+
+        job.useScm(GitScm.class)
+                .url(repoUrl)
+                .credentials(USERNAME)
+                .calculateChangelog("origin", BRANCH_NAME);
+        job.save();
+
+        b = job.startBuild();
+        b.shouldSucceed();
+        changesUrl = b.url("changes");
+
+        assertThat(
+                visit(changesUrl).getPageSource(),
+                Matchers.containsRegexp("/user/jenkins-ath/", Pattern.MULTILINE)
+        );
+
+        job.configure();
+        job.useScm(GitScm.class).commitAuthorInChangelog();
+
+        job.save();
+
+        b = job.startBuild();
+        b.shouldSucceed();
+        changesUrl = b.url("changes");
+
+        assertThat(
+                visit(changesUrl).getPageSource(),
+                Matchers.containsRegexp("/user/other-email/", Pattern.MULTILINE)
+        );
+    }
+
+    @Test
+    public void advanced_clone_test() {
+        final String TEST_INITIAL_COMMIT = "Initial commit";
+        final String TEST_COMMIT1_MESSAGE = "First commit";
+        final String TEST_COMMIT2_MESSAGE = "Second commit";
+        Build b;
+
+        GitRepo repo = buildGitRepo();
+        repo.changeAndCommitFoo(TEST_COMMIT1_MESSAGE);
+        repo.changeAndCommitFoo(TEST_COMMIT2_MESSAGE);
+        repo.transferToDockerContainer(host, port);
+
+        GitScm gitScm = job.useScm(GitScm.class)
+                .url(repoUrl)
+                .credentials(USERNAME);
+        job.addShellStep("git log");
+        job.save();
+
+        b = job.startBuild();
+        b.shouldSucceed();
+        String console = b.getConsole();
+        assertThat(
+                console,
+                Matchers.containsRegexp(TEST_INITIAL_COMMIT, Pattern.MULTILINE)
+        );
+        assertThat(
+                console,
+                Matchers.containsRegexp(TEST_COMMIT1_MESSAGE, Pattern.MULTILINE)
+        );
+        assertThat(
+                console,
+                Matchers.containsRegexp(TEST_COMMIT2_MESSAGE, Pattern.MULTILINE)
+        );
+
+        job.configure();
+        GitScm.AdvancedClone behaviour = gitScm.advancedClone()
+                .checkShallowClone(true)
+                .setNumDepth("2");
+        job.save();
+
+        b = job.startBuild();
+        b.shouldSucceed();
+        console = b.getConsole();
+        assertThatFail(console, TEST_INITIAL_COMMIT, "Shallow clone depth is 2");
```

Comment:
`assertThat(console, not(ANY_MATCHER_HERE))`?

Reply:
Changed